### PR TITLE
EBP-33: As a Solace Developer, I can access CCSMP cache constants and enums

### PR DESCRIPTION
**Pull request by oodigie**
**Changes:**
- EBP-33: As a Solace Developer, I can access CCSMP cache constants and enums

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,6 +66,14 @@ go generate .
 ```
 This will generate subcode_generated.go from the given solClient.h
 
+And to generate ccsmp_cache_session_prop_generated.go and the other ccsmp props/enums from the given solClient.h and solCache.h, run this:
+```
+export SOLCACHE_H=/path/to/repo/lib/linux/include/solclient/solCache.h
```
Comment (gszol) [inline CONTRIBUTING.md:71]: I think there isn't "/linux/" in that path.
And we'll have to change the instructions here to include solCache.h in the copy and set the new env var:
https://sol-jira.atlassian.net/wiki/spaces/CPD/pages/2868052115/Golang+Process+and+Tools#Updating-CCSMP

I updated my script just now.

**Comment by sdewilde** [comment]
I'd love to see a quick test showing that we can access the defaults that were added

File: CONTRIBUTING.md (hunk 1 above)
Comment (oodigie) [reply CONTRIBUTING.md:71]: Okay. I have made the update to remove the `/linux` from the path template.

**Review by gszol — APPROVED** [review]
Looks good to me with the latest doc changes, thank you.

File: internal/ccsmp/ccsmp_cache_session_prop_generated.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	SolClientCacheSessionPropRequestreplyTimeoutMs = C.SOLCLIENT_CACHESESSION_PROP_REQUESTREPLY_TIMEOUT_MS
+	// SolClientCacheSessionPropReplyTo: Deprecated: The reply-to topic for the cache request. It is preferred to leave this property unspecified and let the API use the Session P2P Reply To Topic: ::SOLCLIENT_SESSION_PROP_P2PINBOX_IN_USE.
+	SolClientCacheSessionPropReplyTo = C.SOLCLIENT_CACHESESSION_PROP_REPLY_TO
+	// SolClientCacheSessionPropDefaultMaxMsgs: The default maximum number of message to retrieve from the cache for any one topic is 1.
+	SolClientCacheSessionPropDefaultMaxMsgs = C.SOLCLIENT_CACHESESSION_PROP_DEFAULT_MAX_MSGS
+	// SolClientCacheSessionPropDefaultMaxAge: Default maximum age is no maximum age (set to zero).
+	SolClientCacheSessionPropDefaultMaxAge = C.SOLCLIENT_CACHESESSION_PROP_DEFAULT_MAX_AGE
+	// SolClientCacheSessionPropDefaultRequestreplyTimeoutMs: The default timeout (in milliseconds) to wait for a response from the cache is 10000 (10 seconds).
+	SolClientCacheSessionPropDefaultRequestreplyTimeoutMs = C.SOLCLIENT_CACHESESSION_PROP_DEFAULT_REQUESTREPLY_TIMEOUT_MS
```
Comment (sdewilde) [inline internal/ccsmp/ccsmp_cache_session_prop_generated.go:42]: EBP-33 outlines that the acceptance criteria should confirm these values are as expected.  Either thought an automated test or post your manual test results in the jira.

Comment (sdewilde) [reply internal/ccsmp/ccsmp_cache_session_prop_generated.go:42]: If you've eyeballed them and printed them out for example I think that's okay on the manual side.  I wanted to make sure we'd at least confirmed that much

**Pull request merged**